### DISCUSSION TITLE
Failure to customize cross sections in cell "taper_cross_section" in v8.10

### DISCUSSION THREAD

**Discussion by Fengchenghao1996** [gdsfactory/gdsfactory#2898] [Q&A] [answered]
Hi, I am migrating my codes from v7 to v8. I found one issue when using gf.components.taper_cross_section. It seems it cannot get the layer I created in the layermap and will return errors such as "45 is not a valid LAYER". Here are simple codes to repeat this error. The function is to create a taper. 
```
from gdsfactory.technology import LayerMap

import gdsfactory as gf

class LayerMap0(LayerMap):
    NWG = (174, 0)
gds1 = gf.Component()

Layer = LayerMap0

s0 = gf.Section(width=1.0, offset=0, layer=Layer.NWG,  name = "core", port_names=("o1", "o2") )
xc_sin = gf.CrossSection(sections=[s0],radius = 35.0)

s1 = gf.Section(width=0.7, offset=0, layer=Layer.NWG,  name = "core",port_names=("o1", "o2"))
xc_sin1 = gf.CrossSection(sections=[s1],radius = 35.0)

taper1 = gds1 << gf.components.taper_cross_section(cross_section1 = xc_sin1,
                                           cross_section2 = xc_sin,
                                           length=10)
gds1.plot()
```
The errors I get is:
"
ValueError: 45 is not a valid LAYER
"

Such codes work well on the v7 version (The layer map coding should be revised, of course). Could someone help me debug and see if the errors are from my side? Thanks.

**Comment by Fengchenghao1996** [comment]
Additional comment:

It seems the error will not occur for certain layer numbers. For example, if I change NWG = (1,0), then there will be no errors at all. Therefore, I suspect that the layer setting is incorrect in either my settings or gdsfactory settings.

**Comment by joamatab** [comment, marked as answer]
Hi,

i believe we fixed this in the latest version of gdsfactory,

can you try running after doing `pip install gdsfactory --upgrade` ?

**Reply by Fengchenghao1996** [reply]
Hi Joamatab,

I tested the codes on v8.20 and found no issues. Thanks for your team's input!

Best,
Chenghao
